src/generic_core/uproxy_core.ts

```diff
@@ -181,8 +183,13 @@ export class uProxyCore implements uproxy_core_api.CoreApi {
    */
   public logout = (networkInfo :social.SocialNetworkInfo) : Promise<void> => {
     var networkName = networkInfo.name;
-    var userId = networkInfo.userId;
-    var network = social_network.getNetwork(networkName, userId);
+    if (networkInfo.userId) {
+      var userId = networkInfo.userId;
```

Comment:
Cool. You could also use `let` for `network` (basically, there's almost no reason to use `var` from ES6 onwards - good habit to get into).